lighthouse-core/test/gather/gatherers/seo/font-size-test.js

```diff
@@ -36,53 +60,249 @@ describe('Font size gatherer', () => {
-        analyzedFailingNodesData: [{
+    };
+
+    const artifact = await fontSizeGather.afterPass({driver});
+    const expectedFailingTextLength = smallText.trim().length;
+    const expectedVisitedTextLength = bigText.trim().length + expectedFailingTextLength;
+    const expectedTotalTextLength = failingText.trim().length + expectedVisitedTextLength;
+    const expectedAnalyzedFailingTextLength = expectedFailingTextLength;
+
+    expect(artifact).toEqual({
+      analyzedFailingTextLength: expectedAnalyzedFailingTextLength,
+      failingTextLength: expectedFailingTextLength,
+      visitedTextLength: expectedVisitedTextLength,
+      totalTextLength: expectedTotalTextLength,
+      analyzedFailingNodesData: [
+        {
           fontSize: 10,
           node: bodyNode,
           textLength: expectedFailingTextLength,
-        }],
+        },
+      ],
+    });
+  });
+
+  describe('#computeSelectorSpecificity', () => {
+    const compute = FontSizeGather.computeSelectorSpecificity;
+
+    it('should handle basic selectors', () => {
+      expect(compute('h1')).toEqual(1);
+      expect(compute('h1 > p > span')).toEqual(3);
+    });
+
+    it('should handle class selectors', () => {
+      expect(compute('h1.foo')).toEqual(11);
+      expect(compute('h1 > p.other.yeah > span')).toEqual(23);
+    });
+
+    it('should handle ID selectors', () => {
+      expect(compute('h1#awesome.foo')).toEqual(111);
+      expect(compute('h1 > p.other > span#the-text')).toEqual(113);
+    });
+
+    it('should cap the craziness', () => {
+      expect(compute('h1.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o.p')).toEqual(91);
+    });
+  });
+
+  describe('#getEffectiveFontRule', () => {
+    const createProps = props => Object.entries(props).map(([name, value]) => ({name, value}));
+    const createStyle = ({properties, id}) => ({
+      styleSheetId: id,
+      cssProperties: createProps(properties),
+    });
+    const createRule = ({style, selectors, origin}) => ({
+      style,
+      origin,
+      selectorList: {selectors: selectors.map(text => ({text}))},
+    });
+
+    let inlineStyle;
+    let matchedCSSRules;
+    let inherited;
+
+    beforeEach(() => {
+      const fontRule = createRule({
+        origin: 'regular',
+        selectors: ['html body *', '#main'],
+        style: createStyle({id: 123, properties: {'font-size': '1em'}}),
       });
+      const userAgentRule = createRule({
+        origin: 'user-agent',
+        selectors: ['body'],
+        style: createStyle({properties: {'font-size': 12}}),
+      });
+
+      inlineStyle = createStyle({id: 1, properties: {'font-size': '1em'}});
+      matchedCSSRules = [{matchingSelectors: [1], rule: fontRule}];
+      inherited = [{matchedCSSRules: [{matchingSelectors: [0], rule: userAgentRule}]}];
+    });
+
+    it('should identify inline styles', () => {
+      const result = FontSizeGather.getEffectiveFontRule({inlineStyle});
+      expect(result).toMatchInlineSnapshot(`
+Object {
```

Comment:
*IMO*, it seems like
a) it would be shorter and easier to read as an inline object literal and an expect.toEqual
b) if reading isn't important, then could just be a regular snapshot (we have shorter ones :)
c) if these were going to occasionally update, maybe there's an advantage for snapshots' ease of use, but these shouldn't ever change
rather than bringing in prettier just for this feature

Reply:
I like snapshot testing. I can't figure out why one would want to do it inline, though... Seems creating the initial snapshot + approving updates to the snapshot become a lot more menial tasks.

Reply:
@Hoten the inline snapshot is still managed automatically (that's why we needed to update jest and add a prettier dependency), the benefit is you just get to see the result right next to the test instead of a separate file

Reply:
Snapshot testing all the way! ♥️

Reply:
I think inline snapshots introduce benefits to 1) testing consistency and 2) dev ergonomics

**Consistency**

The alternative we're weighing this against is `expect.toEqual(obj)`. But `obj` could be an object literal or it could just constructed some other way.. (`expected = JSON.parse(JSON.stringify(thing)); expected.change = 42`) 

Adopting inline snapshots brings consistency to this pattern and prioritizes a declarative view of the expected object. I think this encourages more reviewable tests.

**Ergonomics**

First time writing the assertion, just do `expect(result).toMatchInlineSnapshot()` and the test will be updated with the current result. When making any changes that affect the expected value, updating them is just a matter of typing `u`.  

To be fair, we don't spend _a lot_ of time writing expecting values, but these workflow improvements make authoring tests faster and easier. (Which encourages more tests).

------

(I looked across our usage of deepStrictEqual and the large majority are very small expected objects... like 1-3 lines tall)

TBH I think toEqual() continues to make sense if the obj literal is like 1-4ish lines tall. From 5-15 lines tall, inline snapshot I think wins for creating/updating advantages.  And past 15 lines moving to a non-inline snapshot makes the most sense.

wdyt?

Reply:
> First time writing the assertion, just do expect(result).toMatchInlineSnapshot() and the test will be updated with the current result. When making any changes that affect the expected value, updating them is just a matter of typing u.

this is actually why I think snapshots are great. I agree they might be overkill for smaller objects but should we really care? It's easier to always snapshot test just to make it consistent for contributers? The downside of snapshot testing is that your not always seeing what the result is going to be.

with snapshot:
```
const obj = myCoolFunction();
expect(obj).toMatchInlineSnapshot()
```

without:
```
const obj = myCoolFunction();
assert.deepStrictEqual(obj, { a': 'a', b: { c: 'c'}})
```

nevertheless I still think snapshot testing are great for testing artifacts & audit results